sui/operators.js

```diff
@@ -0,0 +1,283 @@
+        options: {
+            showContent: true,
+        },
+    });
+
+    // Extract and return the gas collector cap ID
+    const gasCollectorCapId = gasCollectorCapObject.data.content.fields.value.fields.id.id;
+    return gasCollectorCapId;
+}
+
+// TODO: Fix `InvalidPublicFunctionReturnType { idx: 0 } in command 0` error
+async function collectGas(keypair, client, config, chain, args, options) {
+    const [amount] = args;
+    const receiver = options.receiver || keypair.toSuiAddress();
+    const gasServiceConfig = config.sui.contracts.GasService;
+    const operatorsConfig = config.sui.contracts.Operators;
+
+    if (!gasServiceConfig) {
+        throw new Error('Gas service package not found.');
+    }
+
+    if (!operatorsConfig) {
+        throw new Error('Operators package not found.');
+    }
+
+    const operatorId = operatorsConfig.objects.Operators;
+    const gasCollectorCapId = await getGasCollectorCapId(client, gasServiceConfig, operatorsConfig);
+    const operatorCapId = await findOwnedObjectId(client, keypair.toSuiAddress(), `${operatorsConfig.address}::operators::OperatorCap`);
+
+    printInfo('GasCollectorCap', gasCollectorCapId);
+    printInfo('OperatorCap', operatorCapId);
+    printInfo('Operators', operatorId);
+
+    const tx = new Transaction();
+
+    const borrowedCap = tx.moveCall({
+        target: `${operatorsConfig.address}::operators::borrow_cap`,
+        arguments: [tx.object(operatorId), tx.object(operatorCapId), tx.pure(bcs.Address.serialize(gasCollectorCapId).toBytes())],
+        typeArguments: [`${gasServiceConfig.address}::gas_service::GasCollectorCap`],
+    });
+
+    tx.moveCall({
+        target: `${gasServiceConfig.address}::gas_service::collect_gas`,
+        arguments: [tx.object(gasServiceConfig.objects.GasService), borrowedCap, tx.pure.address(receiver), tx.pure.u64(amount)],
+    });
+
+    const receipt = await broadcast(client, keypair, tx);
+    printInfo('Gas collected', receipt.digest);
+}
+
+// TODO: Fix `InvalidPublicFunctionReturnType { idx: 0 } in command 0` error
+async function refund(keypair, client, config, chain, args, options) {
+    const [messageId] = args;
+    const amount = args.amount;
+    const receiver = options.receiver || keypair.toSuiAddress();
+
+    const gasServiceConfig = config.sui.contracts.GasService;
+    const operatorsConfig = config.sui.contracts.Operators;
+
+    if (!gasServiceConfig) {
+        throw new Error('Gas service package not found.');
+    }
+
+    if (!operatorsConfig) {
+        throw new Error('Operators package not found.');
+    }
+
+    const operatorId = operatorsConfig.objects.Operators;
+    const gasCollectorCapId = await getGasCollectorCapId(client, gasServiceConfig, operatorsConfig);
+    const operatorCapId = await findOwnedObjectId(client, keypair.toSuiAddress(), `${operatorsConfig.address}::operators::OperatorCap`);
+
+    const tx = new Transaction();
+
+    const borrowedCap = tx.moveCall({
```

Comment:
```suggestion
    const [borrowedCap] = tx.moveCall({
```
do you need to explicitly destructure?